jujuclient/file.go

```diff
@@ -547,6 +543,27 @@ func updateModels(
 	return nil
 }
 
+// SetModels implements ModelUpdater.
+func (s *store) SetModels(controllerName string, models map[string]ModelDetails) error {
+	if err := ValidateControllerName(controllerName); err != nil {
+		return errors.Trace(err)
+	}
+
```

Comment:
Yep, just to be clear: it should happen before you get to updateModels, because updateModels requires that you take the lock and read the file. That sort of thing should be avoided if possible (and in this case, it is).